### DISCUSSION TITLE
是否考虑支持webhook？

### DISCUSSION THREAD

**Discussion by chivehao** [bangumi/server#445] [Ideas]
## 背景

我们[Ikaros](https://ikaros.run)目前已经实现了通过[Ikaros的番组计划插件plugin-bgmtv](https://docs.ikaros.run/docs/plugins/plugin-bgmtv)在用户看完一个剧集的时候，同步更新番组计划上的进度，
这是从 Ikaros => 番组计划

## 需求
Ikaros希望也能感知到番组计划的用户收藏进度更新，具体就是用户在点击看完某个剧集后，能接受到一个通知。
这是从 番组计划 => Ikaros

所以希望番组计划可以支持webhook。

当然Ikaros也是可以通过定时查询番组计划用户的收藏来感知到用户的收藏进度，但是这个接口查询频率不好把握，更重要的是这不优雅。

## 场景

用户在其它地方看了这个剧集，通过番组计划网页或者app点格子，番组计划通过用户配置的webhook，发送通知。

**Comment by trim21** [comment]
如果你愿意开发的话倒是可以支持这个功能，现在后端有kafka和debezium，应用可以按需订阅mysql binlog。

**Reply by trim21** [reply]
不一定用go，其他语言也可以。

https://github.com/bangumi/purge-old-cache/blob/10e9cf026cb907e9e63b425ba6acb274d558944c/app/main.py#L45

**Reply by chivehao** [reply]
> 不一定用go，其他语言也可以。

python我也不会😂，暂时不考虑参与了，以后有机会一定参与。
